hmt_escrow/job.py

```diff
@@ -335,12 +335,14 @@ def launch(self, pub_key: bytes) -> bool:
         # Use factory to deploy a new escrow contract.
         trusted_handlers = [addr for addr, priv_key in self.multi_credentials]
 
-        txn_success = self._create_escrow(trusted_handlers)
+        txn = self._create_escrow(trusted_handlers)
 
-        if not txn_success:
+        if not txn["txn_succeeded"]:
             raise Exception("Unable to create escrow")
 
-        job_addr = self._last_escrow_addr()
+        tx_receipt = txn["tx_receipt"]
+        events = self.factory_contract.events.Launched().processReceipt(tx_receipt)
```

Comment:
I think this will throw an exception because on polygon the logs contain 2 events, and one of them isn't "Launched" event

Reply:
Could you possibly check that?

Reply:
Checked, we have one event for currently supported networks: polygon, polygon-mumbai, rinkeby